src/twisted/web/test/test_wsgi.py

```diff
@@ -1941,6 +1943,91 @@ def cbRendered(ignored):
         return d
 
 
+    @inlineCallbacks
+    def test_pendingRequestOperationsRespectConnectionLoss(self):
+        """
+        If there are pending transport writes after
+        C{Request.notifyFinish} is called, the transport should
+        not be written to. The test issues a request to a wsgi
+        app, which schedules the transport to disconnect. All writes
+        to the transport are blocked until after the disconnect
+        finishes, forcing the write-after-disconnect behaviour.
+        """
+
+        self.enableThreads()
+        trackedRequest = [None]
+
+        def applicationFactory():
+            """
+            A wsgi application which schedules a request
+            disconnect, and returns a single response chunk.
+            """
+            def application(environ, startResponse):
+                startResponse('200 OK', [])
+
+                self.reactor.callFromThread(
+                    trackedRequest[0].connectionLost,
+                    Failure(ConnectionLost("No more connection")))
+
+                yield b'single chunk response'
+
+            return application
+
+        def requestFactory(*args, **kwargs):
+            # A factory which returns a twisted.web.server.Request
```

Comment:
minor comment: why not use docstrings here?

Reply:
Done.